### PR TITLE
fix(tensorrt_yolo): change the CMakeLists.txt folders for compiling error on Jetson AGX-Orin

### PR DESCRIPTION
**Pull request by NilaySener**
## Description

Jetson AGX-Orin uses Jetpack 5.1.0 which includes:
- CUDA 11.4.14
- TensorRT 8.4.0 Early Access
- cuDNN 8.3.2

These library versions were giving deprecation errors while compiling some packages:
- tensorrt_yolo
- lidar_apollo_instance_segmentation
- lidar_centerpoint
- traffic_light_ssd_fine_detector
- traffic_light_classifier

To fix this issue, compiler flag added to CMakeLists.txt. Should we add this situation to documantation? I opened an[ issue](https://github.com/autowarefoundation/autoware-documentation/issues/166) in autoware-documentation for this case.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,8 @@ find_package(autoware_cmake REQUIRED)
 autoware_package()
 
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
-
+set(DEPRECATION_FLAG "-Wno-error=deprecated-declarations")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}"
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:9]:
```suggestion
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}")
```

File: perception/lidar_apollo_instance_segmentation/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +26,8 @@ else()
   set(CUDA_AVAIL OFF)
 endif()
 
+
+
```
Comment (wep21) [inline perception/lidar_apollo_instance_segmentation/CMakeLists.txt:30]:
```suggestion
```

Comment (wep21) [reply perception/lidar_apollo_instance_segmentation/CMakeLists.txt:30]: unnecessary change

File: perception/lidar_centerpoint/CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,8 @@ find_package(autoware_cmake REQUIRED)
 autoware_package()
 
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
-
+set(DEPRECATION_FLAG "-Wno-error=deprecated-declarations")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}"
```
Comment (wep21) [inline perception/lidar_centerpoint/CMakeLists.txt:9]:
```suggestion
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}")
```

File: perception/tensorrt_yolo/CMakeLists.txt [hunk 4]
```diff
@@ -7,7 +7,8 @@ autoware_package()
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 find_package(OpenCV REQUIRED)
-
+set(DEPRECATION_FLAG "-Wno-error=deprecated-declarations")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}"
```
Comment (wep21) [inline perception/tensorrt_yolo/CMakeLists.txt:11]:
```suggestion
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}")
```

File: perception/traffic_light_classifier/CMakeLists.txt [hunk 5]
```diff
@@ -5,6 +5,9 @@ find_package(autoware_cmake REQUIRED)
 autoware_package()
 
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
+
+set(DEPRECATION_FLAG "-Wno-error=deprecated-declarations")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}"
```
Comment (wep21) [inline perception/traffic_light_classifier/CMakeLists.txt:10]:
```suggestion
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}")
```

File: perception/traffic_light_ssd_fine_detector/CMakeLists.txt [hunk 6]
```diff
@@ -7,7 +7,8 @@ autoware_package()
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 find_package(OpenCV REQUIRED)
-
+set(DEPRECATION_FLAG "-Wno-error=deprecated-declarations")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}"
```
Comment (wep21) [inline perception/traffic_light_ssd_fine_detector/CMakeLists.txt:11]:
```suggestion
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEPRECATION_FLAG}")
```

**Comment by xmfcx** [comment]
@wep21 can you approve it if it is ok now?

**Comment by wep21** [comment]
@xmfcx As I mentioned in [the PR](https://github.com/autowarefoundation/autoware.universe/pull/1472), I would like to merge it instead of this PR.

**Pull request closed without merging**